test/common/filter/config_discovery_impl_test.cc

```diff
@@ -454,6 +459,310 @@ TEST_F(FilterConfigDiscoveryImplTest, TerminalFilterInvalid) {
             scope_.counter("extension_config_discovery.http_filter.foo.config_reload").value());
 }
 
+// Listener ECDS test
+// Using TlsInspector listener filter in these tests.
+class ListenerFilterConfigDiscoveryImplTest : public FilterConfigDiscoveryTestBase {
+public:
+  ListenerFilterConfigDiscoveryImplTest() {
+    filter_config_provider_manager_ = std::make_unique<ListenerFilterConfigProviderManagerImpl>();
+  }
+  ~ListenerFilterConfigDiscoveryImplTest() override {
+    factory_context_.thread_local_.shutdownThread();
+  }
+
+  DynamicFilterConfigProviderPtr<Network::ListenerFilterFactoryCb>
+  createProvider(std::string name, bool warm, bool default_configuration,
+                 bool last_filter_config = true) {
```

Comment:
Good catch! ACK

Reply:
The latest change have generalized the based test class to support HTTP filter, TCP/UDP listener filter to avoid boiler plates in test code. With that, the parameter is added back.  PTAL.